tests/integration_tests/functional/test_rate_limiter.py

```diff
@@ -158,95 +161,51 @@ def _check_tx_rate_limiting(test_microvm, guest_ips, host_ips):
-    )
-    iperf_out = _process_iperf_output(iperf_out)[1]
-
-    rate_limit_bytes_achieved = 1000 * float(iperf_out)
-    rate_limit_bytes_expected = rate_limit_bps
-
-    # Assert on the bytes expected and achieved with rate limiting on; we are
-    # expecting a difference no bigger than MAX_RATE_LIMIT_BYTES_DIFF
-    assert (
-            _get_difference(
-                rate_limit_bytes_achieved,
-                rate_limit_bytes_expected
-            )
-            < MAX_BYTES_DIFF_PERCENTAGE
-    )
+    # Sanity check that bandwidth with no rate limiting is at least double
+    # than the one expected when rate limiting is in place.
+    assert _get_percentage_difference(rate_no_limit_kbps, expected_kbps) > 100
+
+    # Second step: check bandwith when rate limiting is on.
+    _check_tx_bandwidth(test_microvm, guest_ips[1], host_ips[1], expected_kbps)
 
     # Third step: get the number of bytes when rate limiting is on and there is
     # an initial burst size from where to consume.
 
+    print("Run guest TX iperf with exact burst size")
     # Use iperf to obtain the bandwidth when there is burst to consume from.
-    iperf_cmd = '{} -c {} -n{} -f KBytes'.format(
-        IPERF_BINARY,
-        host_ips[2],
-        BURST_SIZE
-    )
-    iperf_out = _run_iperf_on_guest(test_microvm, iperf_cmd, guest_ips[2])
-    # iperf will give variable number of output lines depending on how much
-    # time it took to send the amount specified.
-    burst_bw_first_time_achieved = _process_iperf_output(iperf_out)[1]
-
-    # The second time we use iperf to send bytes we need to see that the burst
-    # was consumed and that the transmit rate is now equal to the rate limit.
-    # We are sending the amount of bytes that can be sent in 1 sec with rate
-    # limiting enabled.
-    iperf_cmd = '{} -c {} -n{} -f KBytes'.format(
+    iperf_cmd = '{} -c {} -n {} -f KBytes -w {} -N'.format(
         IPERF_BINARY,
         host_ips[2],
-        rate_limit_bps
+        BURST_SIZE,
+        IPERF_TCP_WINDOW
     )
     iperf_out = _run_iperf_on_guest(test_microvm, iperf_cmd, guest_ips[2])
-    iperf_out_time, iperf_out_bw = _process_iperf_output(iperf_out)
-
-    # Test that the bandwidth we obtained first time is at least as two times
-    # higher than the one obtained when rate limiting is on.
-    assert _get_difference(burst_bw_first_time_achieved, iperf_out_bw) > 100
-
-    # Test that the bandwidth we obtained second time is at least two times
-    # lower than the one obtained when no rate limiting is in place.
-    assert _get_difference(rate_no_limit_bytes, iperf_out_bw) > 100
-
-    burst_consumed_time_achieved = iperf_out_time
-    # We expect it to take around 1 sec now.
-    burst_consumed_time_expected = 1
-
-    assert (
-            _get_difference(
-                burst_consumed_time_achieved,
-                burst_consumed_time_expected
-            )
-            < MAX_TIME_DIFF
-    )
+    print(iperf_out)
```

Comment:
nit: I'd include the `print`s in `_run_iperf_on_*` and add a "verbosity" parameter to the `_run_iperf`s

Reply:
I think it'd be superfluous. `Pytest` already has that knob in the form of the `-s` parameter. By default all test stdout is suppressed, running with `-s` will show prints.

The CI only shows prints in case of failures.